### PR TITLE
LPS-192640 Adding an editable ERC component

### PR DESCRIPTION
**Pull request by oliv-yu**
https://liferay.atlassian.net/browse/LPS-192640 - The ERC of a Blueprint and Element can easily be accessed, edited and copied

These changes add the editable ERC component to SXP Blueprints and Elements toolbar. 
- The validation to check empty/duplicate/formatted ERC is performed within the "Edit ERC" modal. ERC is able to accept alphanumerics and dashes/underscores.
- Readonly elements would not be able to edit ERC
- Since ERC is taking up more of the toolbar and it's possible title might get cut off, tooltips were added to title and ERC

Some screenshots:
Screenshot 2023-09-06 at 4 56 40 PM
Screenshot 2023-09-07 at 4 49 32 PM
Screenshot 2023-09-06 at 4 58 15 PM
Screenshot 2023-09-06 at 4 58 56 PM
Screenshot 2023-09-07 at 4 46 19 PM

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf

**Comment by oliv-yu** [comment]
ci:test:search

**Comment by oliv-yu** [comment]
ci:test:relevant

**Comment by oliv-yu** [comment]
@thektan When trying it out, I had some trouble with using the APIs (I asked Petteri about it [here](https://liferay.atlassian.net/browse/LPS-192640?focusedCommentId=2453516)). Do you mind checking that as well, see if I need to update something or if it's more a backend thing? Thank you!

**Review by thektan — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js [hunk 1]
```diff
@@ -605,6 +609,10 @@ function EditSXPBlueprintForm({
 		setIsTitleAndDescriptionEdited(true);
 	};
 
+	const _handleChangeExternalReferenceCode = (externalReferenceCode) => {
```
[the remark this answers is not in the document]
Comment (thektan) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/edit_sxp_blueprint/EditSXPBlueprintForm.js]: Pushed up a new commit for this change. [b826856](https://github.com/liferay-search/liferay-portal/pull/1065/commits/b82685694c694003c6ed356488cfd1a3966cfbf8)

**Review by thektan — APPROVED** [review]

**Review by peerkar — APPROVED** [review]
@thektan service changes look good to me 👍

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:search

**Comment by thektan** [comment]
ci:test:relevant

**Comment by thektan** [comment]
ci:test:relevant

**Comment by joshchong** [comment]
ci:forward:force

**Comment by joshchong** [comment]
ci:forward:force

**Comment by joshchong** [comment]
ci:forward:force

**Pull request closed without merging**